apis/v1alpha1/collector_webhook.go

```diff
@@ -378,7 +376,7 @@ func (c CollectorWebhook) validateTargetAllocatorConfig(ctx context.Context, r *
 		if subjectAccessReviews, err := c.reviewer.CheckPolicyRules(ctx, r.GetNamespace(), r.Spec.TargetAllocator.ServiceAccount, targetAllocatorCRPolicyRules...); err != nil {
 			return nil, fmt.Errorf("unable to check rbac rules %w", err)
 		} else if allowed, deniedReviews := rbac.AllSubjectAccessReviewsAllowed(subjectAccessReviews); !allowed {
-			return warningsGroupedByResource(deniedReviews), nil
+			return rbac.WarningsGroupedByResource(deniedReviews), nil
```

Comment:
I might be missing here something so take this with a grain of salt.

I thought that an admission warning should be printed if a component is enabled that requires custom role and the operator cannot create it.

Reply:
The idea was to check if the service account of the operator has the needed permissions to create RBAC resources when the application starts. If it has those permissions, we will try to create them when needed (every time we reconcile an OTEL Collector, for instance). Else, we will skip.

So, if you change the permissions to the service account to enable/disable this feature, you will need to restart the OTEL Operator deployment.